ext/drpttracer/drpttracer.c

```diff
@@ -586,14 +562,21 @@ drpttracer_end_tracing(IN void *drcontext, INOUT void *tracer_handle,
 }
 
 DR_EXPORT
-void
+drpttracer_status_t
 drpttracer_destroy_output(IN void *drcontext, IN drpttracer_output_t *output)
 {
-    ASSERT(output != NULL, "trying to free NULL output buffer");
-    ASSERT(output->pt_buf != NULL, "trying to free NULL PT buffer");
+    if (output == NULL) {
+        ASSERT(false, "trying to free NULL output buffer");
+        return DRPTTRACER_ERROR_TRYING_TO_FREE_NULL_OUTPUT;
```

Comment:
No change needed but generally it's not required to add a separate enum value for each situation. Something like INVALID_INPUT would suffice for all these cases.

Reply:
Done.